toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -394,3 +402,70 @@ func addCustomizerRelease(imageChroot *safechroot.Chroot, toolVersion string, bu
 
 	return nil
 }
+
+func handleSELinux(selinuxMode imagecustomizerapi.SELinux, selinuxDisablePackageCheck bool,
+	partitionsCustomized bool, imageChroot *safechroot.Chroot,
+) error {
+	var err error
+
+	// Resolve the default SELinux mode.
+	if selinuxMode == imagecustomizerapi.SELinuxDefault {
+		selinuxMode, err = getCurrentSELinuxMode(imageChroot)
+		if err != nil {
+			return err
+		}
+	}
+
+	// If the partitions were customized, then the grub.cfg file will have been recreated from scratch and therefore
+	// the SELinux args will already be correct and don't need to be updated.
+	if !partitionsCustomized {
+		// Update the SELinux kernel command-line args.
+		err := updateSELinuxCommandLine(selinuxMode, imageChroot)
+		if err != nil {
+			return fmt.Errorf("failed to update SELinux args in grub.cfg:\n%w", err)
+		}
+	}
+
+	if selinuxMode != imagecustomizerapi.SELinuxDisabled {
```

Comment:
The code behind this check is rather lengthy and has a few nested `if` statements as well. Could we:
1. Leave early to reduce the number of indents?
2. Extract the code behind this check into another function?
```suggestion
	if selinuxMode == imagecustomizerapi.SELinuxDisabled {
		return nil
	}
```

I'm mostly interested in the second bit to be honest. If the check stays the way it is but we have that separate function, then it will get cleaner anyway.

Reply:
fixed